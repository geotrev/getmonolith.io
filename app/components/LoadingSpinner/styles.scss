@import "../../global";

#loading-spinner {
  width: 100%;
  display: flex;
  flex-flow: row nowrap;
  justify-content: center;
  padding: 64px;
}
