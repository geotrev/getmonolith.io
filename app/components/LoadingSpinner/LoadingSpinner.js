import React from "react"
import ScaleLoader from "react-spinners/ScaleLoader"

import "./styles.scss"

const SPINNER_COLOR = "#d15ded"

export default function LoadingSpinner() {
  return (
    <div id="loading-spinner">
      <ScaleLoader color={SPINNER_COLOR} />
    </div>
  )
}
