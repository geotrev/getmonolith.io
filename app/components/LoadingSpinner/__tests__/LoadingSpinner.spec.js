import React from "react"
import LoadingSpinner from "../LoadingSpinner"

describe("<LoadingSpinner />", () => {
  it("matches snapshot", () => {
    const wrapper = mount(<LoadingSpinner />)
    expect(wrapper).toMatchSnapshot()
  })
})
