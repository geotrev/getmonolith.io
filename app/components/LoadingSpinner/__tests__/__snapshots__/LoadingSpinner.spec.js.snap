// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingSpinner /> renders 1`] = `
<LoadingSpinner>
  <div
    id="loading-spinner"
  >
    <Loader
      color="#d15ded"
      css=""
      height={35}
      loading={true}
      margin={2}
      radius={2}
      width={4}
    >
      <EmotionCssPropInternal
        __EMOTION_TYPE_PLEASE_DO_NOT_USE__="div"
        css={
          Array [
            "",
          ]
        }
      >
        <div
          className="css-0"
        >
          <EmotionCssPropInternal
            __EMOTION_TYPE_PLEASE_DO_NOT_USE__="div"
            css={
              Object {
                "map": undefined,
                "name": "1z13box",
                "next": Object {
                  "name": "animation-6rlus4",
                  "next": undefined,
                  "styles": "@keyframes animation-6rlus4{
  0% {transform: scaley(1.0)}
  50% {transform: scaley(0.4)}
  100% {transform: scaley(1.0)}
}",
                },
                "styles": "
      background-color: #d15ded;
      width: 4px;
      height: 35px;
      margin: 2px;
      border-radius: 2px;
      display: inline-block;
      animation: animation-6rlus4 1s 0.1s infinite cubic-bezier(0.2, 0.68, 0.18, 1.08);
      animation-fill-mode: both;
    ",
                "toString": [Function],
              }
            }
          >
            <div
              className="css-1z13box"
            />
          </EmotionCssPropInternal>
          <EmotionCssPropInternal
            __EMOTION_TYPE_PLEASE_DO_NOT_USE__="div"
            css={
              Object {
                "map": undefined,
                "name": "1vjwbje",
                "next": Object {
                  "name": "animation-6rlus4",
                  "next": undefined,
                  "styles": "@keyframes animation-6rlus4{
  0% {transform: scaley(1.0)}
  50% {transform: scaley(0.4)}
  100% {transform: scaley(1.0)}
}",
                },
                "styles": "
      background-color: #d15ded;
      width: 4px;
      height: 35px;
      margin: 2px;
      border-radius: 2px;
      display: inline-block;
      animation: animation-6rlus4 1s 0.2s infinite cubic-bezier(0.2, 0.68, 0.18, 1.08);
      animation-fill-mode: both;
    ",
                "toString": [Function],
              }
            }
          >
            <div
              className="css-1vjwbje"
            />
          </EmotionCssPropInternal>
          <EmotionCssPropInternal
            __EMOTION_TYPE_PLEASE_DO_NOT_USE__="div"
            css={
              Object {
                "map": undefined,
                "name": "1j7ry5d",
                "next": Object {
                  "name": "animation-6rlus4",
                  "next": undefined,
                  "styles": "@keyframes animation-6rlus4{
  0% {transform: scaley(1.0)}
  50% {transform: scaley(0.4)}
  100% {transform: scaley(1.0)}
}",
                },
                "styles": "
      background-color: #d15ded;
      width: 4px;
      height: 35px;
      margin: 2px;
      border-radius: 2px;
      display: inline-block;
      animation: animation-6rlus4 1s 0.30000000000000004s infinite cubic-bezier(0.2, 0.68, 0.18, 1.08);
      animation-fill-mode: both;
    ",
                "toString": [Function],
              }
            }
          >
            <div
              className="css-1j7ry5d"
            />
          </EmotionCssPropInternal>
          <EmotionCssPropInternal
            __EMOTION_TYPE_PLEASE_DO_NOT_USE__="div"
            css={
              Object {
                "map": undefined,
                "name": "dho9nr",
                "next": Object {
                  "name": "animation-6rlus4",
                  "next": undefined,
                  "styles": "@keyframes animation-6rlus4{
  0% {transform: scaley(1.0)}
  50% {transform: scaley(0.4)}
  100% {transform: scaley(1.0)}
}",
                },
                "styles": "
      background-color: #d15ded;
      width: 4px;
      height: 35px;
      margin: 2px;
      border-radius: 2px;
      display: inline-block;
      animation: animation-6rlus4 1s 0.4s infinite cubic-bezier(0.2, 0.68, 0.18, 1.08);
      animation-fill-mode: both;
    ",
                "toString": [Function],
              }
            }
          >
            <div
              className="css-dho9nr"
            />
          </EmotionCssPropInternal>
          <EmotionCssPropInternal
            __EMOTION_TYPE_PLEASE_DO_NOT_USE__="div"
            css={
              Object {
                "map": undefined,
                "name": "gdz7da",
                "next": Object {
                  "name": "animation-6rlus4",
                  "next": undefined,
                  "styles": "@keyframes animation-6rlus4{
  0% {transform: scaley(1.0)}
  50% {transform: scaley(0.4)}
  100% {transform: scaley(1.0)}
}",
                },
                "styles": "
      background-color: #d15ded;
      width: 4px;
      height: 35px;
      margin: 2px;
      border-radius: 2px;
      display: inline-block;
      animation: animation-6rlus4 1s 0.5s infinite cubic-bezier(0.2, 0.68, 0.18, 1.08);
      animation-fill-mode: both;
    ",
                "toString": [Function],
              }
            }
          >
            <div
              className="css-gdz7da"
            />
          </EmotionCssPropInternal>
        </div>
      </EmotionCssPropInternal>
    </Loader>
  </div>
</LoadingSpinner>
`;
